### ISSUE TITLE
 double display of the ip on netwok inventory

### ISSUE THREAD

**Issue by trasher** [glpi-project/glpi-inventory-plugin#106]
the ip no longer disappears on the printers but there is the double display of the ip. It's almost good

![image](https://user-images.githubusercontent.com/68054991/162402001-63a8d939-b09d-4dc5-bd1f-fb77258304f1.png)
![image](https://user-images.githubusercontent.com/68054991/162403486-3d095136-502e-48af-a5e8-e16e44b74c94.png)

Example :

ETH0
![image](https://user-images.githubusercontent.com/68054991/162402282-3b90cac0-8b54-48ea-b76e-baa1183571bf.png)

MANAGEMENT
![image](https://user-images.githubusercontent.com/68054991/162402777-fef069ab-efce-457c-b9f2-647304809883.png)

_Originally posted by @steph461 in https://github.com/glpi-project/glpi-inventory-plugin/issues/53#issuecomment-1092624176_

**Comment by leruetkins** [comment]
Confirm for glpi 10 and glpi-agent 1.0 with import rules by default:

![изображение](https://user-images.githubusercontent.com/15270519/165068576-c1c03561-68d2-4019-815c-2dc487830153.png)

![изображение](https://user-images.githubusercontent.com/15270519/165068992-c33ddab1-adde-4376-8444-606a5dc87bfc.png)

**Comment by trasher** [comment]
I'm not able to reproduce. Could you try with latest GLPI nightly?

**Comment by leruetkins** [comment]
Yes, I have installed 10.0-9b92c4c.tar.gz  (2022-04-23 00:25:27 UTC) but have the same result. I think it's problem with management port:
![изображение](https://user-images.githubusercontent.com/15270519/165073278-4c8e9aab-df66-465d-8dfe-c3ac00377f24.png)

![изображение](https://user-images.githubusercontent.com/15270519/165073403-ec28aa94-775c-4658-a22c-fee4e0aa83ce.png)
![изображение](https://user-images.githubusercontent.com/15270519/165073484-02c6b44d-bdee-43f7-90e1-134c9f544e9e.png)

Two ports have one ip address.

**Comment by steph461** [comment]
3 times  : 
![image](https://user-images.githubusercontent.com/68054991/165483081-3f92a1a9-ced9-49b6-bf36-cd81561de480.png)
![image](https://user-images.githubusercontent.com/68054991/165483528-6288f300-815e-4e42-914b-e14b830b7cbc.png)

glpi 10.0

**Comment by steph461** [comment]
Its possibble fusion the two interface etho ? I think it's a duplicate
The good data is "Eth0" color green, i think

**Comment by WebGregGit** [comment]
I suspect that the switch was inventoried with fusioninventory, and now you have switched to glpi inventory. If so - remove the switch and inventory it again.

**Comment by trasher** [comment]
> I suspect that the switch was inventoried with fusioninventory, and now you have switched to glpi inventory. If so - remove the switch and inventory it again.

It's possible, yes; that would explain why I cannot reproduce. 
If that solve the issue, this is maybe a migration bug.

**Comment by leruetkins** [comment]
I don't use Fusion Inventory, I use native inventory only.

**Comment by steph461** [comment]
I tried to delete the printers and inventory same bug
he must have another data in trance which poses problem I think

**Comment by steph461** [comment]
1 . Delete two printers ; I start from 0

2. Scan discovery network, no bug display
![image](https://user-images.githubusercontent.com/68054991/167818933-a202c20f-e855-4142-9b4f-d0f70d1ede8d.png)

3. Scan SNMP bug display
![image](https://user-images.githubusercontent.com/68054991/167819843-b006153e-24a4-433c-ab6a-9c659552da14.png)
![image](https://user-images.githubusercontent.com/68054991/167820440-488998af-97c8-43e0-ad69-2f065772f973.png)

**Comment by steph461** [comment]
there is better by deleting and inventorying the printers but it always displays the two ips
Eth0 is displayed only once and it's good

Management and Eth0 display show ips in preview

**Comment by steph461** [comment]
Computer : 
If that can help. When I inventory a computer that has multiple IPs. it correctly displays the two different ips. For me its ok
![image](https://user-images.githubusercontent.com/68054991/167823951-1a366d1a-ddb3-4c0b-aac3-a4cac2ca2351.png)
![image](https://user-images.githubusercontent.com/68054991/167824403-f45c8ad0-5ba1-4862-ad9c-2789f8cbd1a0.png)

Printer : 
The function management doesnt exist for the computer and its normal.
![image](https://user-images.githubusercontent.com/68054991/167824602-39a80304-b345-490d-b820-946c8482fe11.png)

@trasher is it possible to say, display the ips that are on ETHO for the printers for example ?

**Comment by brunnersm** [comment]
I have the same problem.
But, my printers don't have management interface.
![image](https://user-images.githubusercontent.com/69863640/172212412-2fa974f8-0ac7-4f8d-a2e3-28772d193154.png)
![image](https://user-images.githubusercontent.com/69863640/172212487-12753cbf-9a1e-4d9d-b19b-d8d7f1e91677.png)
![image](https://user-images.githubusercontent.com/69863640/172212588-eb230c04-4399-4310-be19-8356875f3586.png)

**Comment by leruetkins** [comment]
Is it possible to disable the management port during the import?

**Comment by steph461** [comment]
> Is it possible to disable the management port during the import?

Section Network
the management port can be util for the discovery Switch ( VLAN IP) 
![image](https://user-images.githubusercontent.com/68054991/173578398-60bb0859-92ff-41d1-876d-a03bcf532fd5.png)
![image](https://user-images.githubusercontent.com/68054991/173578595-9d33e9fa-273a-48e1-a8cd-cdd1720176d8.png)

Section printer
Printer section may be handle it differently ?

**Comment by leruetkins** [comment]
I want it to be done automatically. I don't want to delete everything with my hands every time after the discovery.
Also I have noticed that MAC address also double display.
![изображение](https://user-images.githubusercontent.com/15270519/173579620-a80ec1e2-8e8a-46d9-a993-d55aa9b9f570.png)

**Comment by stonebuzz** [comment]
Hi @brunnersm @steph461  @leruetkins 

 For duplicate IP from list view 
can you test this -> https://github.com/glpi-project/glpi/pull/12171

Best regards

**Comment by stonebuzz** [comment]
Hi @brunnersm @steph461 @leruetkins

For duplicate MaAC from list view

can you test this -> https://github.com/glpi-project/glpi/pull/12172

Best regards

**Comment by stonebuzz** [comment]
Hi @brunnersm @steph461 @leruetkins
after applying patches you should need 
this one -> https://github.com/glpi-project/glpi/pull/12176
To prevent error "ambigous colulmn ..."

Best regards

**Comment by leruetkins** [comment]
Hi
I still have problems, see: https://github.com/glpi-project/glpi/pull/12171

**Comment by brunnersm** [comment]
Hi @stonebuzz ,
i applied this fix https://github.com/glpi-project/glpi/pull/12171
solve my problem. I took advantage and applied https://github.com/glpi-project/glpi/pull/12172 and https://github.com/glpi-project/glpi/pull/12176
Thanks

**Comment by brunnersm** [comment]
Hi @stonebuzz ,
Now, after fix, my NetworkEquipment list hidden ips. (printers all OK)
![image](https://user-images.githubusercontent.com/69863640/178038622-d10bb51b-3856-4d58-84d1-778b7acf62c1.png)
![image](https://user-images.githubusercontent.com/69863640/178038729-7d2382e6-d291-465a-ae66-6e0325612a49.png)
![image](https://user-images.githubusercontent.com/69863640/178038791-cc6fa2a9-180d-472d-863b-3fe8a32eb487.png)

**Comment by brunnersm** [comment]
I undid the fix https://github.com/glpi-project/glpi/pull/12171
![image](https://user-images.githubusercontent.com/69863640/178041441-dc5e81ee-ced3-4652-9044-8271d0c12021.png)

**Comment by steph461** [comment]
hello, I'm on vacation, I'll test this patch when I get back

**Comment by stonebuzz** [comment]
No problem @steph461 have a Nice vacation ^^

**Comment by stonebuzz** [comment]
> I undid the fix https://github.com/glpi-project/glpi/pull/12171
> ![image](https://user-images.githubusercontent.com/69863640/178041441-dc5e81ee-ced3-4652-9044-8271d0c12021.png)
> 
> And now, printers NOT ok. 2 ips listed.

Maybe printer and network equipment are handle network port differently, i need to do more test next week, i give you a feedback ASAP Stay tuned, have a good weekend  =)

**Comment by leruetkins** [comment]
> > Hi
> > I still have problems, see: [glpi-project/glpi#12171](https://github.com/glpi-project/glpi/pull/12171)
> 
> Please retry with glpi 10.0.2 (latest) and all patches
> 
> Best regard

Hi
I updated glpi to 10.0.2 and applied all pathces, but have the same situation with printers:
![изображение](https://user-images.githubusercontent.com/15270519/178192116-23d0cb3c-f867-4b93-b465-4ee17996dac2.png)
![изображение](https://user-images.githubusercontent.com/15270519/178192243-bda01a47-f2cf-4ce8-973b-12c9a55b268e.png)
![изображение](https://user-images.githubusercontent.com/15270519/178192347-9005409d-ae85-4853-88c5-9bd2d0a9ee7e.png)

**Comment by leruetkins** [comment]
Some printers look like this:
![изображение](https://user-images.githubusercontent.com/15270519/178236734-a837a026-75f0-41a0-86c7-69538d6d3d21.png)
Another printers look like this:
![изображение](https://user-images.githubusercontent.com/15270519/178236888-42f0f346-5e2a-49cd-b3ea-33cd1c1c00f0.png)

**Comment by stonebuzz** [comment]
> Some printers look like this: ![изображение](https://user-images.githubusercontent.com/15270519/178236734-a837a026-75f0-41a0-86c7-69538d6d3d21.png) Another printers look like this: ![изображение](https://user-images.githubusercontent.com/15270519/178237204-42994642-9fe0-4126-be5b-813895951d86.png)

I imagine that on the first screenshot you have an IP displayed in the column (from printer list)
 and on the second screenshot you have nothing?

**Comment by stonebuzz** [comment]
Can you post here XML from this printer (with only port management) ?

Best regards

**Comment by leruetkins** [comment]
Yes, you are right.
Sure:
[inv.zip](https://github.com/glpi-project/glpi-inventory-plugin/files/9083482/inv.zip)

**Comment by stonebuzz** [comment]
after injecting  XML from ```netDiscovery``` then ```netInventory```

Printer have : 
- 1 port management (internal)
- 1 port Ethernet (eth0)

![image](https://user-images.githubusercontent.com/7335054/178255105-1cd7a660-074b-4eb4-8a06-19961799f5ef.png)

From Printer list IP et MAC are no longer duplicated

Before applying this patches

1. https://github.com/glpi-project/glpi/pull/12171
2. https://github.com/glpi-project/glpi/pull/12172
3. https://github.com/glpi-project/glpi/pull/12176

![image](https://user-images.githubusercontent.com/7335054/178255370-5262de56-b3f1-476c-ade3-0b9c55859493.png)

After applying patches :

![image](https://user-images.githubusercontent.com/7335054/178255755-9c105397-b16a-479b-9603-83d98255d423.png)

IP and MAC from "port management" are no longer displayed

All is OK for this computer

**Comment by stonebuzz** [comment]
@leruetkins 
from ```netDiscovery``` step, inventory only have "port management"

![image](https://user-images.githubusercontent.com/7335054/178257225-7bd90e58-06e0-4318-82e2-470cfc8999a2.png)

```NetworkPortEthernet``` comes from ```NetInventory```

If Printer Doesn't have ```NetworkPortEthernet```  probably you haven't done the ```netInventory``` yet

**Comment by stonebuzz** [comment]
Hi @brunnersm 

> Hi @stonebuzz , Now, after fix, my NetworkEquipment doesn't list the ips. (printers all OK) 

Can you past here XML inventory file from your switch ?

**Comment by leruetkins** [comment]
I had to delete an existing printer and do netdiscovery and then netInventory. After that eth0 is shown. Is it mean that I have to delete all my printers and add all them again?

Now it looks better, but I still have 2 links:
![изображение](https://user-images.githubusercontent.com/15270519/178260438-8e4cb6aa-0855-41b5-a24c-211a173f9ebe.png)

**Comment by brunnersm** [comment]
Hi @stonebuzz ,
Yes, below:
[CCIRU_SW1.zip](https://github.com/glpi-project/glpi-inventory-plugin/files/9083970/CCIRU_SW1.zip)

**Comment by stonebuzz** [comment]
@brunnersm 
thanks.

After importing your inventory file, I only have the  ports management, nothing in the "Network Port" panel.

![image](https://user-images.githubusercontent.com/7335054/178266291-3c047dba-2bb0-414e-8a61-6914a99a9d89.png)

is it the same for you

**Comment by leruetkins** [comment]
> 

Yes, It helped
![изображение](https://user-images.githubusercontent.com/15270519/178266605-f3100115-96bb-4f4d-bdd8-5fab42f1e131.png)
Now I have "Ethernet" instead "eth0", but it is not critical for me.
![изображение](https://user-images.githubusercontent.com/15270519/178266739-525f810d-1162-430e-8c1b-85ef46191e31.png)

**Comment by brunnersm** [comment]
@stonebuzz ,
Yes, same thing.
I waiting fix for this.

**Comment by stonebuzz** [comment]
> > 
> 
> Yes, It helped, but MAC duplicated still presented: ![изображение](https://user-images.githubusercontent.com/15270519/178266605-f3100115-96bb-4f4d-bdd8-5fab42f1e131.png) Now I have "Ethernet" instead "eth0", but it is not critical for me. ![изображение](https://user-images.githubusercontent.com/15270519/178266739-525f810d-1162-430e-8c1b-85ef46191e31.png)

For MAC have you test

https://github.com/glpi-project/glpi/pull/12172 
and
https://github.com/glpi-project/glpi/pull/12176

**Comment by leruetkins** [comment]
> > > 
> > 
> > 
> > Yes, It helped, but MAC duplicated still presented: ![изображение](https://user-images.githubusercontent.com/15270519/178266605-f3100115-96bb-4f4d-bdd8-5fab42f1e131.png) Now I have "Ethernet" instead "eth0", but it is not critical for me. ![изображение](https://user-images.githubusercontent.com/15270519/178266739-525f810d-1162-430e-8c1b-85ef46191e31.png)
> 
> For MAC have you test
> 
> [glpi-project/glpi#12172](https://github.com/glpi-project/glpi/pull/12172) and [glpi-project/glpi#12176](https://github.com/glpi-project/glpi/pull/12176)

Yes, I applied all these patches.
Also I tried do net discovery and netinventory for existing printer and tried to delete it comletle and do net discovery and netinventory again. It doesn't help.

**Comment by stonebuzz** [comment]
> @stonebuzz , Yes, same thing. I waiting fix for this. All ports are in the xml but they don't show up.

Can you open new issue, is not same problem as duplicate IP / MAC

Best regards

**Comment by stonebuzz** [comment]
> > > > 
> > > 
> > > 
> > > Yes, It helped, but MAC duplicated still presented: ![изображение](https://user-images.githubusercontent.com/15270519/178266605-f3100115-96bb-4f4d-bdd8-5fab42f1e131.png) Now I have "Ethernet" instead "eth0", but it is not critical for me. ![изображение](https://user-images.githubusercontent.com/15270519/178266739-525f810d-1162-430e-8c1b-85ef46191e31.png)
> > 
> > 
> > For MAC have you test
> > [glpi-project/glpi#12172](https://github.com/glpi-project/glpi/pull/12172) and [glpi-project/glpi#12176](https://github.com/glpi-project/glpi/pull/12176)
> 
> Yes, I applied all these patches. Also I tried do net discovery and netinventory for existing printer and tried to delete it comletle and do net discovery and netinventory again. It doesn't help.

Can you past here XML inventory fiel for NPIB9C331

**Comment by stonebuzz** [comment]
@brunnersm 

> @stonebuzz , Yes, same thing. I waiting fix for this. All ports are in the xml but they don't show up.

it appears that the ports do not appear in GLPI because the ```IFNUMBER``` does not exist in the XML

```xml
        <PORT>
          <IFINOCTETS>0</IFINOCTETS>
          <IFINTERNALSTATUS>1</IFINTERNALSTATUS>
          <IFNAME>MEth0/0/1</IFNAME>
          <IFSPEED>10000000</IFSPEED>
          <IFSTATUS>2</IFSTATUS>
          <IFTYPE>6</IFTYPE>
          <IP>192.168.234.106</IP>
          <IPS>
            <IP>192.168.234.106</IP>
          </IPS>
          <MAC>28:a6:db:3f:96:e0</MAC>
        </PORT>
```

If i add it manually

```xml
        <PORT>
          <IFINOCTETS>0</IFINOCTETS>
          <IFINTERNALSTATUS>1</IFINTERNALSTATUS>
          <IFNAME>MEth0/0/1</IFNAME>
          <IFSPEED>10000000</IFSPEED>
          <IFSTATUS>2</IFSTATUS>
          <IFNUMBER>35</IFNUMBER>
          <IFTYPE>6</IFTYPE>
          <IP>192.168.234.106</IP>
          <IPS>
            <IP>192.168.234.106</IP>
          </IPS>
          <MAC>28:a6:db:3f:96:e0</MAC>
        </PORT>
```

Port is correctly handled by inventory

![image](https://user-images.githubusercontent.com/7335054/178272027-5c5742b7-c696-4460-b498-f996b1c1c440.png)

Are you using the latest version of the agent

If yes, please open issue here -> https://github.com/glpi-project/glpi-agent

I think it is due to the manufacturer (Huawei)

Best regards

**Comment by leruetkins** [comment]
> 

Hi

<?xml version="1.0" encoding="UTF-8" ?>
<REQUEST>
  <CONTENT>
    <DEVICE>
      <CARTRIDGES>
        <TONERBLACK>WARNING</TONERBLACK>
      </CARTRIDGES>
      <INFO>
        <COMMENTS>HP ETHERNET MULTI-ENVIRONMENT,SN:PHBLL8Q9K9,FN:3W9391C,SVCID:28302,PID:HP LaserJet MFP M426dw</COMMENTS>
        <ID>0</ID>
        <IPS>
          <IP>192.168.244.59</IP>
        </IPS>
        <MAC>f4:39:09:b9:c3:31</MAC>
        <MANUFACTURER>Hewlett-Packard</MANUFACTURER>
        <MEMORY>256</MEMORY>
        <MODEL>HP LaserJet MFP M426dw</MODEL>
        <NAME>NPIB9C331</NAME>
        <RAM>256</RAM>
        <SERIAL>PHBLL8Q9K9</SERIAL>
        <TYPE>PRINTER</TYPE>
        <UPTIME>3 hours, 08:52.56</UPTIME>
      </INFO>
      <PAGECOUNTERS>
        <DUPLEX>118</DUPLEX>
        <TOTAL>241785</TOTAL>
      </PAGECOUNTERS>
      <PORTS>
        <PORT>
          <IFDESCR>LOOPBACK</IFDESCR>
          <IFINERRORS>0</IFINERRORS>
          <IFINOCTETS>4843155</IFINOCTETS>
          <IFINTERNALSTATUS>1</IFINTERNALSTATUS>
          <IFLASTCHANGE>0.00 seconds</IFLASTCHANGE>
          <IFMTU>1500</IFMTU>
          <IFNAME>LOOPBACK</IFNAME>
          <IFNUMBER>1</IFNUMBER>
          <IFOUTERRORS>0</IFOUTERRORS>
          <IFOUTOCTETS>4843269</IFOUTOCTETS>
          <IFSPEED>0</IFSPEED>
          <IFSTATUS>1</IFSTATUS>
          <IFTYPE>24</IFTYPE>
        </PORT>
        <PORT>
          <IFDESCR>Ethernet</IFDESCR>
          <IFINERRORS>0</IFINERRORS>
          <IFINOCTETS>106154154</IFINOCTETS>
          <IFINTERNALSTATUS>1</IFINTERNALSTATUS>
          <IFLASTCHANGE>0.00 seconds</IFLASTCHANGE>
          <IFMTU>1500</IFMTU>
          <IFNAME>Ethernet</IFNAME>
          <IFNUMBER>2</IFNUMBER>
          <IFOUTERRORS>0</IFOUTERRORS>
          <IFOUTOCTETS>5102434</IFOUTOCTETS>
          <IFSPEED>10000000</IFSPEED>
          <IFSTATUS>1</IFSTATUS>
          <IFTYPE>6</IFTYPE>
          <IP>192.168.244.59</IP>
          <IPS>
            <IP>192.168.244.59</IP>
          </IPS>
          <MAC>f4:39:09:b9:c3:31</MAC>
        </PORT>
        <PORT>
          <IFDESCR>wifi0</IFDESCR>
          <IFINERRORS>0</IFINERRORS>
          <IFINOCTETS>104009671</IFINOCTETS>
          <IFINTERNALSTATUS>1</IFINTERNALSTATUS>
          <IFLASTCHANGE>8.62 seconds</IFLASTCHANGE>
          <IFMTU>1500</IFMTU>
          <IFNAME>wifi0</IFNAME>
          <IFNUMBER>3</IFNUMBER>
          <IFOUTERRORS>0</IFOUTERRORS>
          <IFOUTOCTETS>0</IFOUTOCTETS>
          <IFSPEED>10000000</IFSPEED>
          <IFSTATUS>1</IFSTATUS>
          <IFTYPE>6</IFTYPE>
          <MAC>74:40:bb:75:5f:a5</MAC>
        </PORT>
        <PORT>
          <IFDESCR>wifiUAP</IFDESCR>
          <IFINERRORS>0</IFINERRORS>
          <IFINOCTETS>0</IFINOCTETS>
          <IFINTERNALSTATUS>1</IFINTERNALSTATUS>
          <IFLASTCHANGE>8.68 seconds</IFLASTCHANGE>
          <IFMTU>1500</IFMTU>
          <IFNAME>wifiUAP</IFNAME>
          <IFNUMBER>4</IFNUMBER>
          <IFOUTERRORS>0</IFOUTERRORS>
          <IFOUTOCTETS>16859</IFOUTOCTETS>
          <IFSPEED>10000000</IFSPEED>
          <IFSTATUS>1</IFSTATUS>
          <IFTYPE>6</IFTYPE>
          <MAC>74:40:bb:75:5f:a5</MAC>
        </PORT>
      </PORTS>
    </DEVICE>
    <MODULEVERSION>5.1</MODULEVERSION>
    <PROCESSNUMBER>1</PROCESSNUMBER>
  </CONTENT>
  <DEVICEID>foo</DEVICEID>
  <QUERY>SNMPQUERY</QUERY>
</REQUEST>

**Comment by stonebuzz** [comment]
> > 
> 
> Hi
> 
> ```
> <?xml version="1.0" encoding="UTF-8" ?>
> <REQUEST>
>   <CONTENT>
>     <DEVICE>
>       <CARTRIDGES>
>         <TONERBLACK>WARNING</TONERBLACK>
>       </CARTRIDGES>
>       <INFO>
>         <COMMENTS>HP ETHERNET MULTI-ENVIRONMENT,SN:PHBLL8Q9K9,FN:3W9391C,SVCID:28302,PID:HP LaserJet MFP M426dw</COMMENTS>
>         <ID>0</ID>
>         <IPS>
>           <IP>192.168.244.59</IP>
>         </IPS>
>         <MAC>f4:39:09:b9:c3:31</MAC>
>         <MANUFACTURER>Hewlett-Packard</MANUFACTURER>
>         <MEMORY>256</MEMORY>
>         <MODEL>HP LaserJet MFP M426dw</MODEL>
>         <NAME>NPIB9C331</NAME>
>         <RAM>256</RAM>
>         <SERIAL>PHBLL8Q9K9</SERIAL>
>         <TYPE>PRINTER</TYPE>
>         <UPTIME>3 hours, 08:52.56</UPTIME>
>       </INFO>
>       <PAGECOUNTERS>
>         <DUPLEX>118</DUPLEX>
>         <TOTAL>241785</TOTAL>
>       </PAGECOUNTERS>
>       <PORTS>
>         <PORT>
>           <IFDESCR>LOOPBACK</IFDESCR>
>           <IFINERRORS>0</IFINERRORS>
>           <IFINOCTETS>4843155</IFINOCTETS>
>           <IFINTERNALSTATUS>1</IFINTERNALSTATUS>
>           <IFLASTCHANGE>0.00 seconds</IFLASTCHANGE>
>           <IFMTU>1500</IFMTU>
>           <IFNAME>LOOPBACK</IFNAME>
>           <IFNUMBER>1</IFNUMBER>
>           <IFOUTERRORS>0</IFOUTERRORS>
>           <IFOUTOCTETS>4843269</IFOUTOCTETS>
>           <IFSPEED>0</IFSPEED>
>           <IFSTATUS>1</IFSTATUS>
>           <IFTYPE>24</IFTYPE>
>         </PORT>
>         <PORT>
>           <IFDESCR>Ethernet</IFDESCR>
>           <IFINERRORS>0</IFINERRORS>
>           <IFINOCTETS>106154154</IFINOCTETS>
>           <IFINTERNALSTATUS>1</IFINTERNALSTATUS>
>           <IFLASTCHANGE>0.00 seconds</IFLASTCHANGE>
>           <IFMTU>1500</IFMTU>
>           <IFNAME>Ethernet</IFNAME>
>           <IFNUMBER>2</IFNUMBER>
>           <IFOUTERRORS>0</IFOUTERRORS>
>           <IFOUTOCTETS>5102434</IFOUTOCTETS>
>           <IFSPEED>10000000</IFSPEED>
>           <IFSTATUS>1</IFSTATUS>
>           <IFTYPE>6</IFTYPE>
>           <IP>192.168.244.59</IP>
>           <IPS>
>             <IP>192.168.244.59</IP>
>           </IPS>
>           <MAC>f4:39:09:b9:c3:31</MAC>
>         </PORT>
>         <PORT>
>           <IFDESCR>wifi0</IFDESCR>
>           <IFINERRORS>0</IFINERRORS>
>           <IFINOCTETS>104009671</IFINOCTETS>
>           <IFINTERNALSTATUS>1</IFINTERNALSTATUS>
>           <IFLASTCHANGE>8.62 seconds</IFLASTCHANGE>
>           <IFMTU>1500</IFMTU>
>           <IFNAME>wifi0</IFNAME>
>           <IFNUMBER>3</IFNUMBER>
>           <IFOUTERRORS>0</IFOUTERRORS>
>           <IFOUTOCTETS>0</IFOUTOCTETS>
>           <IFSPEED>10000000</IFSPEED>
>           <IFSTATUS>1</IFSTATUS>
>           <IFTYPE>6</IFTYPE>
>           <MAC>74:40:bb:75:5f:a5</MAC>
>         </PORT>
>         <PORT>
>           <IFDESCR>wifiUAP</IFDESCR>
>           <IFINERRORS>0</IFINERRORS>
>           <IFINOCTETS>0</IFINOCTETS>
>           <IFINTERNALSTATUS>1</IFINTERNALSTATUS>
>           <IFLASTCHANGE>8.68 seconds</IFLASTCHANGE>
>           <IFMTU>1500</IFMTU>
>           <IFNAME>wifiUAP</IFNAME>
>           <IFNUMBER>4</IFNUMBER>
>           <IFOUTERRORS>0</IFOUTERRORS>
>           <IFOUTOCTETS>16859</IFOUTOCTETS>
>           <IFSPEED>10000000</IFSPEED>
>           <IFSTATUS>1</IFSTATUS>
>           <IFTYPE>6</IFTYPE>
>           <MAC>74:40:bb:75:5f:a5</MAC>
>         </PORT>
>       </PORTS>
>     </DEVICE>
>     <MODULEVERSION>5.1</MODULEVERSION>
>     <PROCESSNUMBER>1</PROCESSNUMBER>
>   </CONTENT>
>   <DEVICEID>foo</DEVICEID>
>   <QUERY>SNMPQUERY</QUERY>
> </REQUEST>
> ```

Hi @leruetkins 
same here

same MAC adresses for two differents ```NetworkPort````(Name and Port number)

![image](https://user-images.githubusercontent.com/7335054/178448980-37115d3c-9507-4e56-af6b-2712cfc2e308.png)

So GLPI display twice

**Comment by stonebuzz** [comment]
Hi

Found better way to 

- handle port management 
- prevent duplicate IP / MAC 
- prevent IP display issue from networkEquipement
hi @leruetkins 


Please revert this patch : 
https://github.com/glpi-project/glpi/pull/12171
https://github.com/glpi-project/glpi/pull/12172
https://github.com/glpi-project/glpi/pull/12176

And follow the guide here  ->   https://github.com/glpi-project/glpi-inventory-plugin/issues/189

I closed this issue.

Best regards
